### ISSUE TITLE
Change default constructor method

### ISSUE THREAD

**Issue by Fuzzyma** [svgdotjs/svg.js#714]
When no function is passed for `config.create` the default function is used:

```js
function() {
      this.constructor.call(this, SVG.create(config.create))
    }
```

this function calls the `SVG.Element` constructor with a node generated by `SVG.create` and the string passed to `config.create`. E.g. `SVG.create('g')`.

However, this behavior has a problem when adopting existing nodes.
Until now, the adopter creates a new instance of - e.g. - `rect` which in turn creates a node `rect`. But we want to adopt a node, so in the next step the references to this node are replaced by the node we want to adopt. Imo its not that great to create objects you wont need.

So my change to the above function will make passing a node to the - in this case - rect-constructor possible so that it uses the right node right from the start:

```
function(node) {
      this.constructor.call(this, node || SVG.create(config.create))
    }
```

This simple change will make adopting nodes more easily:

`new SVG.Rect(node)` is now possible and can be used in the adopter. The lines 
```js
  // ensure references
  element.type  = node.nodeName
  element.node  = node
  node.instance = element
```
are not needed anymore.

As a bonus also this lines can be removed because the SVG.Doc constructor will take care of that: 

```js
  // SVG.Class specific preparations
  if (element instanceof SVG.Doc)
    element.namespace().defs()
```

We could also add 
```js
  // pull svgjs data from the dom (getAttributeNS doesn't work in html5)
  element.setData(JSON.parse(node.getAttribute('svgjs:data')) || {})
```
to the `SVG.Element` constructor which makes adopting of elements almost trivial.

However, the last step needs to be discussed. It might add overhead for the case when creating new nodes (which surely dont have the `svgjs:data` attribute)

**Comment by RmiTtro** [comment]
This seem all good to me :+1:. As for the last step, I think doing something like this should limit the overhead:

```javascript
if(node.hasAttribute('svg:data') {
   element.setData(JSON.parse(node.getAttribute('svgjs:data')) || {})
}
```

**Comment by Fuzzyma** [comment]
I will create a benchmark for this case and look how it affects the creation of new elements. The adopter will stay in any way - it will just get more slim. So having the handling of the svgjs:data attribute there wouldnt be much of a deal.
It just feels more clean to have it in the constructor

**Comment by Fuzzyma** [comment]
@RmiTtro i created a bench where I integrated your code into the SVG.Element constructor (because thats the constructor which is eventually called by any object on creation). I did not added it to the default constructor because in that case it would need to be added to every constructor not using the default.
So I think its better in the SVG.Element's constructor.

My test showed that there are no speed differences between both variants. The 10000 rects test runs as fast as before
